### PR TITLE
Remove flux store usages from utils

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Remove flux store usages from utils. We should probably not have any references to the stores at all in the utils but that's a much larger task than what I'm trying to do here.

@lindy65 can you test to make sure MFA enforcement still works and that clicking on desktop notifications sends you to the right channel?

#### Ticket Link
Part of https://mattermost.atlassian.net/browse/MM-12564

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by lindy65** [comment]
@jwilander - can you please take a look at why the spinmint won't build?

**Comment by lindy65** [comment]
Thanks @jwilander 

Tested:
 - MFA enforcement 
    - works as expected for a user who has MFA turned on already as well as for a user who doesn't and has to set it up at next login. 
 - clicking on desktop notifications sends you to the right channel
    - works as expected - no issues found

File: utils/utils.jsx [hunk 1]
```diff
@@ -122,21 +120,32 @@ export function isSystemAdmin(roles) {
     return false;
 }
 
+export function getTeamRelativeUrl(team) {
```
Comment (sudheerDev) [inline utils/utils.jsx]: Nit: utils/url.jsx  Seems like a is a better place for this.

**Comment by grundleborg** [comment]
@jwilander I think this needs a rebase.

**Pull request merged**